inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/Atan2Function.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.Function;
+
+import java.math.BigDecimal;
+
+/**
+ * Atan2Function
+ * description: atan2(numeric)--returns the arc tangent of a coordinate (numeric1, numeric2).
+ */
+@TransformFunction(names = {"atan2"})
+public class Atan2Function implements ValueParser {
+
+    private ValueParser xParser;
+    private ValueParser yParser;
+
+    public Atan2Function(Function expr) {
+        xParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+        yParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(1));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object xObj = xParser.parse(sourceData, rowIndex, context);
+        Object yObj = yParser.parse(sourceData, rowIndex, context);
+
+        BigDecimal xValue = OperatorTools.parseBigDecimal(xObj);
```

Comment:
Please add null pointer exception check about xObj.

Reply:
Thank you for your mindful review! Here is my little confuse:
What action should happen if null pointer been checked? Throw exceptions like IllegalArgumentException or ArithmeticException, Or just return null as result?
And why only these 2 methods need null pointer exception check？I know that Java Math Functions need params ( not null, otherwise would cause compile error) in raw data type which can transfer to DOUBLE, then dose other trigonometric function also need check?

Reply:
If the parameter can not be parsed, then return null.
For general business, it is acceptable to miss field, but miss data.

Reply:
For some strict validation scenarios, adopting a dirty data shunting scheme after throwing an exception is a better choice.

Reply:
> If the parameter can not be parsed, then return null. For general business, it is acceptable to miss field, but miss data.

Understand. Then I'll change to return null if the parameter can not be parsed.